tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -12773,6 +12834,30 @@ HLSLMaxRecordsAttr *ValidateMaxRecordsAttributes(Sema &S, Decl *D,
                          A.getAttributeSpellingListIndex());
 }
 
+HLSLWaveSizeAttr *ValidateWaveSizeAttributes(Sema &S, Decl *D,
+                                             const AttributeList &A) {
+  // make sure we are in an appropriate shader model
+  const auto *SM =
+      hlsl::ShaderModel::GetByName(S.getLangOpts().HLSLProfile.c_str());
```

Comment:
I think I need to leave the c_str() because it won't just accept a string. GetByName requires a const char *.

Reply:
Gah! That's... wrong. We should fix that.

You can change GetByName to take a StringRef and it shouldn't require any additional changes since `const char *` implicitly converts to StringRef.